src/services/skill-service.ts

```diff
@@ -95,6 +95,10 @@ export const deleteById = async (id: number) => {
   if (skill === null) {
     throw new CustomError("Skill not found", 400)
   }
+  const relatedRecords = await SkillRepository.constrainedSkills(id)
+  if (relatedRecords.projectSkills.length > 0 || relatedRecords.projectMemberSkills.length > 0) {
```

Comment:
Then do the checking here, if (existingProjectSkillCount > 0 || existingProjectMemberSkillCount > 0)

Reply:
done :3